### ISSUE TITLE
Heuristic for inline v. block layout @math{...}

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/curriculum#1361]
The original package had a bug that made this unusably large, which is why we rendered all `@math{...}` directives as text when using slides. I fixed that bug this morning, and `md2googleslides` now converts real mathjax into a usable image.  🎉

The problem is that Google slides _does not support images as inline elements_. That means you can't stick a picture in the middle of a sentence, and have it exist as part of the text. Google will instead render the text without the image, and stick the image on top as a block-layout element that essentially gets its own line.

99% of the time we use `@math{...}`, it's part of a sentence and we want it to be inline. So our current solution (avoiding mathjax on slides)  is actually the right thing to do.

But @retabak has correctly identified that sometimes the math is so complex that it _can't_ be converted to text. This is especially true in the combinatorics lessons, which use stuff like this:

![image](https://github.com/bootstrapworld/curriculum/assets/77126/3b0a8c55-ad39-48f5-a0de-b735a705c4b4)

In these instances, we want `@math{...}` to generate block-layout images in Google slides. The easy option is to add another directive (e.g. `@block-math{...}`). But @ds26gte came up with the idea of maybe auto-detecting when the math is "complex enough", and having the build script guess at when to use MathJax and when to avoid it.

We need to crowdsource some examples of `@math` that we know we want stuck on its own line. The only ones that really come to mind are like the image above (fractions with non-trivial numerator and/or denominators).

Are there any others?

**Comment by retabak** [comment]
First thing that comes to my mind is the distance formula in the [distance formula lesson.](https://bootstrapworld.org/materials/spring2023/en-us/lessons/distance-formula/index.shtml?pathway=algebra-pyret)

Maybe the formula for slope in [this lesson](https://bootstrapworld.org/materials/spring2023/en-us/lessons/functions-can-be-linear/index.shtml?pathway=algebra-pyret)? (I think that falls into the example you cited, above.)

Probably also some stuff in the Algebra 2 pathway? 

I actually don't think there are that many instances where we would need this... but who knows what the future holds?

**Comment by schanzer** [comment]
Ooooh, great call on the distance formula!

The Algebra 2 pathway uses exponents and subscripts, which I _think_ work well in text-only form. I could certainly see our heuristic being "uses either subscripts, or nontrivial fractions"

**Comment by schanzer** [comment]
Potentially. But I wouldn't use that as a heuristic. If we want to have 1/2 used as part of a sentence, making all fractions block-elements will be a nightmare.

That's why I think it's important to specify only _complex fractions_

**Comment by schanzer** [comment]
@ds26gte so the proposal is: "if a `@math{}` element uses either (fractions _and_ divs) OR (square root _and_ exponents), use blockmath?"

That works for me as a first draft, though I expect we'll need to refine this over time

**Comment by ds26gte** [comment]
Yes, you read the disjunction of patterns right, and I do expect we will occasionally add new patterns that will need this special mathjax treatment.

**Comment by schanzer** [comment]
Awesome - let's implement this for `@math` now, and see how it plays over time. Fortunately it's really easy to tweak as we go forward.
